packages/suite/src/components/suite/SecurityCheck/useReportDeviceCompromised.ts

```diff
@@ -26,34 +32,37 @@ const useCommonData = () => {
 };
 
 const useReportRevisionCheck = () => {
-    const { device } = useDevice();
     const commonData = useCommonData();
-
-    const revisionCheck = isDeviceAcquired(device)
-        ? device.authenticityChecks?.firmwareRevision
-        : null;
-    const isRevisionCheckError = revisionCheck && !revisionCheck.success;
-    const revisionCheckError = isRevisionCheckError ? revisionCheck.error : null;
+    const revisionCheckError = useSelector(selectFirmwareRevisionCheckError);
 
     useEffect(() => {
-        if (!isRevisionCheckError) return;
-        reportCheckFail('revision', { ...commonData, revisionCheckError });
-    }, [commonData, isRevisionCheckError, revisionCheckError]);
+        if (revisionCheckError !== null) {
+            reportCheckFail('revision', { ...commonData, revisionCheckError });
+        }
+    }, [commonData, revisionCheckError]);
 };
 
 const useReportHashCheck = () => {
     const { device } = useDevice();
     const commonData = useCommonData();
 
+    // `errorPayload` must also be extracted, which is why `selectFirmwareHashCheckError` would be impractical
     const hashCheck = isDeviceAcquired(device) ? device.authenticityChecks?.firmwareHash : null;
     const isHashCheckError = hashCheck && !hashCheck.success;
     const hashCheckError = isHashCheckError ? hashCheck.error : null;
     const hashCheckErrorPayload = isHashCheckError ? hashCheck.errorPayload : null;
 
     useEffect(() => {
-        if (!isHashCheckError) return;
+        if (!hashCheckError) return;
+        if (
+            isArrayMember(hashCheckError, skippedHashCheckErrors) &&
+            !isArrayMember(hashCheckError, skippedButReportedHashCheckErrors)
+        ) {
+            return;
+        }
+
         reportCheckFail('hash', { ...commonData, hashCheckError, hashCheckErrorPayload });
-    }, [commonData, hashCheckError, isHashCheckError, hashCheckErrorPayload]);
+    }, [commonData, hashCheckError, hashCheckErrorPayload]);
```

Comment:
:memo: note: I tried passing just `hashCheck` to the useEffect, and declaring `hashCheckError, hashCheckErrorPayload` within the useEffect body.
At first glance it's nicer, as I can early return & avoid the ternaries.
:no_entry_sign: But `hashCheck`obj from redux is referentially unstable, so it Sentry fires repeatedly